exceptions.AssertionError: tftp_process_callback

While running latest devel.

```
ERROR   1418157607.07       community:995   Don't know how to decrypt incoming message for circuit_id 1236378659
ERROR   1418157607.08       community:995   Don't know how to decrypt incoming message for circuit_id 1236378659
ERROR   1418157615.56       community:995   Don't know how to decrypt incoming message for circuit_id 1236378659
ERROR   1418157617.74       community:995   Don't know how to decrypt incoming message for circuit_id 1236378659
ERROR   1418157618.29       community:995   Don't know how to decrypt incoming message for circuit_id 1236378659
ERROR   1418157666.13         handler:376   TFTP[7728 C 82.100.48.2:16967][fc9d324b01ed4d6abe087f83d8df6efe72d4519f.torrent] got ERROR message: code = 0, msg = TFTP[7728 S 80.101.15.232:7759][fc9d324b01ed4d6abe087f83d8df6efe72d4519f.torrent] got ACK with block# 39 while expecting 40
ERROR   1418157667.41         handler:376   TFTP[35520 C 82.100.48.2:16967][e7cc987e1fa3e5eb08f726e3c6cdb3246b08beea.torrent] got ERROR message: code = 0, msg = TFTP[35520 S 80.101.15.232:7759][e7cc987e1fa3e5eb08f726e3c6cdb3246b08beea.torrent] got ACK with block# 4 while expecting 5
ERROR   1418157787.01       community:626   TunnelCommunity: removing exit socket 748261992 too old
ERROR   1418157822.31       community:610   TunnelCommunity: removing relay 3701109915 no activity
ERROR   1418157822.31       community:610   TunnelCommunity: removing relay 2962432130 no activity
ERROR   1418157822.45       community:852   Don't know how to decrypt incoming message for circuit_id 2962432130
ERROR   1418157891.87       community:626   TunnelCommunity: removing exit socket 678288168 too old
ERROR   1418157945.71         handler:376   TFTP[31603 S 82.100.48.2:16967][d6350a624c2b38991c5459dbf93571fa813789ae.torrent] got ERROR message: code = 0, msg = TFTP[31603 C 80.101.15.232:7759][d6350a624c2b38991c5459dbf93571fa813789ae.torrent] Got ACK with block# 28 while expecting 29
ERROR   1418157945.71         handler:376   TFTP[61296 C 82.100.48.2:16967][6b1f4892f501b14e7d2f0b253fc16dce07cee8e4.torrent] got ERROR message: code = 0, msg = TFTP[61296 S 80.101.15.232:7759][6b1f4892f501b14e7d2f0b253fc16dce07cee8e4.torrent] got ACK with block# 41 while expecting 42
ERROR   1418157945.71             log:443   Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/Utilities/twisted_thread.py", line 21, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/TFTP/handler.py", line 233, in data_came_in
    self.register_task(u"tftp_process_callback", reactor.callLater(0, self._process_callbacks))
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/taskmanager.py", line 36, in register_task
    assert not self.is_pending_task_active(name), name
exceptions.AssertionError: tftp_process_callback

Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/Utilities/twisted_thread.py", line 21, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/TFTP/handler.py", line 233, in data_came_in
    self.register_task(u"tftp_process_callback", reactor.callLater(0, self._process_callbacks))
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/taskmanager.py", line 36, in register_task
    assert not self.is_pending_task_active(name), name
exceptions.AssertionError: tftp_process_callback
ERROR   1418158041.87       community:626   TunnelCommunity: removing exit socket 2779952415 too old
ERROR   1418158077.56       community:626   TunnelCommunity: removing exit socket 644766113 too old
```

--- a/Tribler/Core/TFTP/handler.py
+++ b/Tribler/Core/TFTP/handler.py
@@ -153,8 +153,16 @@ class TftpHandler(TaskManager):
 
                 self._cleanup_session(key)
 
-        if has_timeout and not self._callback_scheduled:
+        if has_timeout:
+            self._schedule_callback_processing()
+
+    def _schedule_callback_processing(self):
+        """
+        Schedules a task to process callbacks.
+        """
+        if not self._callback_scheduled:
             self.register_task(u"tftp_process_callback", reactor.callLater(0, self._process_callbacks))
+            self._callback_scheduled = True
 
     def _process_callbacks(self):
         """
@@ -229,8 +237,7 @@ class TftpHandler(TaskManager):
                     fd = session.file_data, ei = session.extra_info: cb(a, fn, fd, ei)
                 self._callbacks.append(callback)
 
-        if not self._callback_scheduled:
-            self.register_task(u"tftp_process_callback", reactor.callLater(0, self._process_callbacks))
+        self._schedule_callback_processing()
 
     def _handle_new_request(self, ip, port, packet):
         """ Handles a new request.

